File: py/IEC62325/MarketOperations/ReferenceData/AdjacentCASet.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106 on Thu Dec 28 13:18:10 2023
from typing import Optional

from IEC62325.MarketOperations.ParticipantInterfaces.BidSelfSched import BidSelfSched
from IEC62325.MarketOperations.ReferenceData.SubControlArea import SubControlArea


class AdjacentCASet:
    """
    Groups Adjacent Control Areas.
    """
    def __init__(self) -> None:
        self.loss_percentage: float = 0.0
        self.sub_control_area: Optional[SubControlArea] = SubControlArea()
        self.bid_self_sched: Optional[BidSelfSched] = BidSelfSched()

File: java/IEC62325/MarketOperations/ReferenceData/AdjacentCASet.java
package IEC62325.MarketOperations.ReferenceData;

import IEC61970.Base.Domain.Float;
import IEC61970.Base.Core.IdentifiedObject;
import IEC62325.MarketOperations.ParticipantInterfaces.BidSelfSched;

/**
 * Groups Adjacent Control Areas.
 * @created 03-Jan-2024 2:15:33 PM
 */
public class AdjacentCASet extends IdentifiedObject {

	/**
	 * Loss percentage
	 */
	public Float lossPercentage ;
	public SubControlArea SubControlArea;
	public BidSelfSched BidSelfSched;

	public AdjacentCASet(){

	}
}//end AdjacentCASet